NetKet/Machine/rbm_spin.hpp

```diff
@@ -121,7 +157,12 @@ class RbmSpin : public AbstractMachine<T> {
       lt.V(0).resize(b_.size());
     }
 
-    lt.V(0) = (W_.transpose() * v + b_);
+    if (usewavelets_) {
+      // NOTE: I don't know if the Vector matrix mul is corrrect and if I understood correctly the convolution
+      lt.V(0) = (W_wavelets_ * Wavelets_.transpose() * v + b_);
```

Comment:
I am not sure either, it depends on what Wavelets_ are exactly here.

Reply:
`W_wavelets_` is a vector of length `nw_` (= number of wavelets) and represents the multiplicative weights in front of the wavelets.  `Wavelets_` is a matrix of shape `(nv_, nw_)` so `Wavelets_.transpose()` should be a matrix of shape  `(nw_, nv_)`. In the expression  `W_wavelets_ * Wavelets_.transpose()`, the `*` has to be a pointwise product, i.e.  the row `w, 0 <= w < nw_' of the matrix  `Wavelets_.transpose()` is multiplied by `W_wavelets_(w)`. Thus `W_wavelets_ * Wavelets_.transpose()` is a matrix of shape `(nw_, nv_)`. This seems coherent to me since in my understanding, the number of wavelets `nw_` plays the same role a the number of hidden units `nh_` in the convolutional RBM.